flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -1284,10 +1270,12 @@ private void disposeAllOperatorCoordinators() {
         return coordinatorMap.values();
     }
 
-    private Map<OperatorID, OperatorCoordinatorHolder> createCoordinatorMap() {
+    private Map<OperatorID, OperatorCoordinatorHolder> createCoordinatorMap(
+            ComponentMainThreadExecutor mainThreadExecutor) {
         Map<OperatorID, OperatorCoordinatorHolder> coordinatorMap = new HashMap<>();
         for (ExecutionJobVertex vertex : executionGraph.getAllVertices().values()) {
             for (OperatorCoordinatorHolder holder : vertex.getOperatorCoordinators()) {
+                holder.lazyInitialize(this, mainThreadExecutor);
```

Comment:
Just to double-check: `executionGraph.getAllVertices().values().stream().map(ExecutionJobVertex::getOperatorCoordinators)` contains the same `OperatorCoordinatorHolder` instances as `getAllCoordinators` which was used in `initializeOperatorCoordinators(..)`?

Reply:
Never mind: I didn't see that this method is actually instantiating `coordinatorMap` 😇